wm-system/src/main/kotlin/com/ajaxproject/warehouse/beanpostprocessor/NatsControllerInitializationBeanPostProcessor.kt

```diff
@@ -22,8 +22,10 @@ class NatsControllerInitializationBeanPostProcessor(val connection: Connection)
     ) {
         connection.createDispatcher { message: Message ->
             val parsedData = controller.parser.parseFrom(message.data)
-            val response = controller.handle(parsedData)
-            connection.publish(message.replyTo, response.toByteArray())
+            controller.handle(parsedData)
+                .subscribeOn(Schedulers.boundedElastic())
+                .map { it.toByteArray() }
+                .subscribe { connection.publish(message.replyTo, it) }
```

Comment:
```suggestion
                .map { it.toByteArray() }
                .subscribeOn(Schedulers.boundedElastic())
                .subscribe { connection.publish(message.replyTo, it) }
```

In general, it is better to have subscribe and subscribeOn together, so that later won't be situation when somebody will add some upstream operator that will be executed outside of boundedElastic threads

Reply:
fixed